### PR TITLE
Embed API Client: Java

### PR DESCRIPTION
**Pull request by sgammon**
This changeset adds autogenerated API clients in Java for the *Embed API*, which allows Embedded Menus to be placed on some webpage or other display. This client is low-level and offers advanced access to embeddable configuration details.

Clients are provided for both Maven and Gradle build systems, and are built against API version `v1`.

### CODE REVIEW

File: Java/Embed/Gradle/build.gradle [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    archives sourceJar
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile ([group: 'com.google.api-client', name: 'google-api-client', version: '1.22.0'])
+}
```
Comment (sgammon) [inline Java/Embed/Gradle/build.gradle]: buildfile for Gradle

File: Java/Embed/Gradle/src/main/java/io/bloombox/bloombox/client/EmbedAPI/EmbedAPIRequest.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+   * @param method HTTP Method
+   * @param uriTemplate URI template for the path relative to the base URL. If it starts with a "/"
+   *        the base path from the base URL will be stripped out. The URI template can also be a
+   *        full URL. URI template expansion is done using
+   *        {@link com.google.api.client.http.UriTemplate#expand(String, String, Object, boolean)}
+   * @param content A POJO that can be serialized into JSON or {@code null} for none
+   * @param responseClass response class to parse into
+   */
+  public EmbedAPIRequest(
+      EmbedAPI client, String method, String uriTemplate, Object content, Class<T> responseClass) {
```
Comment (sgammon) [inline Java/Embed/Gradle/src/main/java/io/bloombox/bloombox/client/EmbedAPI/EmbedAPIRequest.java]: example of an `EmbedAPIRequest` in Java

File: Java/Embed/README.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
+# Bloombox: Embed API Client for Java
+
+## Overview
+
+This is a client library bundle using Google Cloud Endpoints. In order to use this API client library in your project, you need to build the library using Maven or Gradle.
+
+### Maven
```
Comment (sgammon) [inline Java/Embed/README.md:7]: readme for Java client - including instructions for Maven and Gradle

File: Java/Embed/build.gradle [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    archives sourceJar
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    compile ([group: 'com.google.api-client', name: 'google-api-client', version: '1.22.0'])
+}
```
Comment (sgammon) [inline Java/Embed/build.gradle:29]: Gradle buildfile

File: Java/Embed/pom.xml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    <groupId>org.sonatype.oss</groupId>
+    <artifactId>oss-parent</artifactId>
+    <version>7</version>
+  </parent>
+
+  <groupId>io.bloombox</groupId>
+  <artifactId>bloombox-client-embedapi</artifactId>
+  <version>v1-1.22.0-SNAPSHOT</version>
+  <name>Embed API v1-1.22.0-SNAPSHOT</name>
+  <packaging>jar</packaging>
```
Comment (sgammon) [inline Java/Embed/pom.xml:13]: POM file for Maven

File: Java/Embed/proguard-google-api-client.txt [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+-dontwarn com.google.api.client.extensions.android.**
+
+# Needed by google-api-client-android when linking against an older platform version
+
+-dontwarn com.google.api.client.googleapis.extensions.android.**
+
+# Needed by google-play-services when linking against an older platform version
+
+-dontwarn com.google.android.gms.**
```
Comment (sgammon) [inline Java/Embed/proguard-google-api-client.txt:23]: proguard configuration

File: Java/Embed/src/main/java/io/bloombox/bloombox/client/EmbedAPI/EmbedAPI.java [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+   * This request holds the parameters needed by the embed server.  After setting any optional
+   * parameters, call the {@link View#execute()} method to invoke the remote operation.
+   *
+   * @param partner
+   * @param location
+   * @param style
+   * @return the request
+   */
+  public View view(java.lang.String partner, java.lang.String location, java.lang.String style) throws java.io.IOException {
+    View result = new View(partner, location, style);
```
Comment (sgammon) [inline Java/Embed/src/main/java/io/bloombox/bloombox/client/EmbedAPI/EmbedAPI.java:250]: method to actually dispatch `embed.view` from Java

File: Java/Embed/src/main/java/io/bloombox/bloombox/client/EmbedAPI/EmbedAPIRequest.java [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+   * @param method HTTP Method
+   * @param uriTemplate URI template for the path relative to the base URL. If it starts with a "/"
+   *        the base path from the base URL will be stripped out. The URI template can also be a
+   *        full URL. URI template expansion is done using
+   *        {@link com.google.api.client.http.UriTemplate#expand(String, String, Object, boolean)}
+   * @param content A POJO that can be serialized into JSON or {@code null} for none
+   * @param responseClass response class to parse into
+   */
+  public EmbedAPIRequest(
+      EmbedAPI client, String method, String uriTemplate, Object content, Class<T> responseClass) {
```
Comment (sgammon) [inline Java/Embed/src/main/java/io/bloombox/bloombox/client/EmbedAPI/EmbedAPIRequest.java:40]: example constructor for an API request in Java - in this case, `EmbedAPIRequest`

**Review by sgammon — COMMENTED** [review]
API client pushed with support for Maven and Gradle.

**Review by tyler-porras — APPROVED** [review]

**Comment by tsamartino** [comment]
Don't know java at all, so there's that.

**Comment by sgammon** [comment]
lol then read some! it's pretty sensible. this stuff is gross though being as it came out of a computer's mind

**Review by RandyStevens — APPROVED** [review]

**Pull request merged**